README.md
# Nerva.Exchange
php based cryptocurrency exchange script

In order to setup Nerva.Exchange's script, your server must have PHP and MySQL installed. You do not have to run any installation scripts for the front end. However, the backend requires daemons running on different screens.

Initially clone the source files to your desired machine. Then edit the files according to your login information. Almost all the pages contain places where you have to insert your database username and password. 

You must also insert the login information for the coin daemons. This script requires you running a pruned (or full) bitcoin core node, and Nerva node and Nerva RPC wallet. 

After doing all these edits on the source codes, you must fix the codes related to Google re-captcha. Once all the settings are done, the script must be working as intended. There are no explanations on the source code as it was designed only by me and I did not need explanations. However, you can fork the project and make changes or add explanations as you wish. Nerva.Exchange was not meant to be an open source project but now it is, so be careful while running it, in terms of security. 

Also you might want to change the location of /order path in order to secure your PHP scripts connecting with your coin daemons. You can set up another folder for your SRC files and make the PHP get the info from there, so web end users will not be able to reach them. 

At last, in order to process orders, you must assign a cron job for the file /order/hiddensrc_exchange/nervaprocessor.php and it must be running every 5-10-60 (or however seconds you want) to check the order status and update the database and process the orders. I used to have 10 seconds cron job that would be checking if the transactions got enough confirmations every 10 seconds.

For any other questions, you can DM me via Discord. Nithronium#8851


changelog.txt
Changelog - Last Updated October 14, 2018 - UTC 13.47
------------------------------------------------------------------------------------

October 14, 2018 --
-------------------

* Order calculation equations are re-written. Now the backend is able to
process big amount of buy requests such as +500 USD ones. (Still there is
order limit tho).

------------------------------------------------------------------------------------
October 13, 2018 --
-------------------
* Input fields on start-order page are now required fields to submit form.
This prevents starting new orders with no XNV or/and BTC address entered.
Also field data are now sent to the server in a different way to prevent
possible SQL injections. 

* Website now uses HTTPS SSL protocol and redirects all HTTP requests to
HTTPS. Users now can safely enter their wallet addresses.

* Added favicon

* Minimum order amount has changed due to some calculation errors made by me.
Now minimum order size is 0.00015 BTC (approximately 90-95 cents).

* Entering non-valid order ID and having a false info displaying order-status
page bug removed. Now if you enter a non-valid order ID, it displays the error
message accordingly.

* Refreshing on the order ID page redirecting to start-order page bug got
removed. Now it redirects to the main page. (This will change in near future
and refreshing will refresh the order status page and will display most
recent order data). 

* Version and release notes link has moved to the left side of main container
 



index.php
<?php session_start(); ?>
<html>
<head>
<link rel="shortcut icon" href="/favicon.ico" />
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script src='https://www.google.com/recaptcha/api.js'></script>
<script>
window.onload = setInterval(function() {
  $.get('https://tradeogre.com/api/v1/orders/BTC-XNV', function(data) {
  obj = JSON.parse(data);
	var nervaprice = Object.keys(obj.sell);
	var nervaprice = nervaprice[0]*1.05;
	var nervaprice = nervaprice.toFixed(8);
	  
	document.getElementById("nerva_price").innerHTML = "1 Nerva = " + Math.round(nervaprice*100000000) + " Satoshi*";
	
	});
},3000);



</script>
  <meta charset="utf-8">
  <meta name="viewport" content="initial-scale=1.0">
  <title>Nerva.Exchange</title>
  <link href="https://fonts.googleapis.com/css?family=Montserrat:400,400|Raleway:100,400,700|Open+Sans:400,300" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="css/standardize.css">
  <link rel="stylesheet" href="css/index-grid.css">
  <link rel="stylesheet" href="css/index.css">
</head>
<body class="body page-index clearfix">
  <div class="wholebody clearfix">
    <div class="header clearfix">
      
      <div class="logo">
        <a href="/"><p class="website-name"><strong>Nerva.Exchange</strong></p></a>
      </div>
      <div class="header-links clearfix">
        <a href="/"><button id="home_button" class="_button _button-1">Home</button></a>
        <p class="text">|</p>
        <a href="about-us.html"><button id="aboutus_button" class="_button _button-2">About Us</button></a>
        <p class="text">|</p>
        <a href="fees.html"><button id="fees_button" class="_button _button-3">Fees</button></a>
      </div>
    </div>
    <div class="container container-1 clearfix">
      <div class="container container-2 clearfix">
        <p class="text text-4">Nerva.Exchange</p>
        <p class="text text-5">© 2018 All Rights Reserved</p>
      </div>
      <div class="container container-3 clearfix">
        <a href="about-us.html"><button class="_button">About Us</button></a>
        <p class="text text-6">|</p>
        <a href="fees.html"><button class="_button">Fees</button></a>
        <p class="text text-7">|</p>
        <a href="terms-conditions.html"><button class="_button">Terms &amp; Conditions</button></a>
        <p class="text text-8">|</p>
        <a href="faq.html"><button class="_button">F.A.Q.</button></a>
      </div>
    </div>
    <div class="body-container">
      <div class="left-container clearfix">
        <p class="text text-9">Exchanging Nerva Made Easy!</p>
        <p class="text text-10"><span>Completely anonym, no registrations, simple exchange</span></p>
        <p class="text text-11"><span>Automatic backend, receive your Nerva in seconds</span></p>
        <p class="text text-12"><font color="#ffffff">No IP restrictions, no KYC requirement</font></p>
		<a href ="changelog.txt"><p class="text text-12"><font color="#ffffff">***Click here to see changelog (Updated with every change)</font></p></a>
		<a href ="v.0.0.1.txt"><p class="text text-12"><font color="#ffffff">Beta Version 0.0.1 - Click to read release notes and roadmap.</font></p></a>
        <p class="text text-13"><font color="#ffffff"><span>* Price is estimated and for information purposes only. The final exchange rate will be determined at the moment the exchange occurs. The final rate could be higher or lower than what it shows on main page.</span></font></p>
        <p class="text text-14"><font color="#ffffff"><span>Currently for Bitcoin transactions, 2 confirmations from the network is required. Exchange will occur after the second confirmation.&nbsp;</span></font></p>
        <p class="text text-15"><font color="#ffffff">Current exchange and transaction fees are; 0.00003 BTC fixed and 4% conversion and Nerva network fee. The fee will be lowered after a week of testing.</font></p>
      </div>
      <div class="right-container clearfix">
        <p id="nerva_price" class="nerva-price">Hold on, I'm getting the price!</p>
		
        <button onclick="location.href='start-order.php'" class="_button _button-8" type="button">Start an Order</button>
        <p class="text text-16"><b>By starting an order, you agree with our terms and conditions of operation.</b></p>
		
		<form action="order-status.php" method="POST" id="check_order_status">
        <p class="text text-17"><?php if ($_SESSION["captcha"] == "false") { echo "Please verify you are human."; $_SESSION["captcha"] = "valid"; }
									elseif($_SESSION["error_status"] == "no_order_with_given_id") { echo "There is no order with given ID."; $_SESSION["error_status"] = "no_error";}
									else { echo "Already have an order?"; } ?>
									
									
									
									
									</p>
        <input class="_input" placeholder="Enter your Order ID here" type="text" name="order_id_check">
        <div class="re_captcha_box"><center><div class="g-recaptcha" data-sitekey="6LfNi3MUAAAAAB5FnElGDVxn6O8yuKZeFFxOBuVJ"></div></center></div>
		<input hidden type="text" name="form-type" value="check-order">
        <button class="_button _button-9" type="submit" form="check_order_status" value="Track Order Status">Track Order Status</button>
		</form>
      </div>
    </div>
  </div>
</body>
</html>

start-order.php
<?php session_start(); ?>
<html>
<head>
<link rel="shortcut icon" href="/favicon.ico" />
<script src='https://www.google.com/recaptcha/api.js'></script>
  <meta charset="utf-8">
  <meta name="viewport" content="initial-scale=1.0">
  <title>Nerva.Exchange | Start Order</title>
  <link href="https://fonts.googleapis.com/css?family=Montserrat:400,400|Open+Sans:400,300|Raleway:400,100,700" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="css/standardize.css">
  <link rel="stylesheet" href="css/start-order-grid.css">
  <link rel="stylesheet" href="css/start-order.css">
</head>
<body class="body page-start-order clearfix">
  <div class="wholebody clearfix">
    <div class="header clearfix">
      <div class="logo">
        <p class="website-name"><strong>Nerva.Exchange</strong></p>
      </div>
      <div class="header-links clearfix">
        <button id="home_button" class="_button _button-1">Home</button>
        <p class="text">|</p>
        <button id="aboutus_button" class="_button _button-2">About Us</button>
        <p class="text">|</p>
        <button id="fees_button" class="_button _button-3">Fees</button>
      </div>
    </div>
    <div class="container container-1 clearfix">
      <div class="container container-2 clearfix">
        <p class="text text-3">Nerva.Exchange</p>
        <p class="text text-4">© 2018 All Rights Reserved</p>
      </div>
      <div class="container container-3 clearfix">
        <button class="_button">About Us</button>
        <p class="text text-5">|</p>
        <button class="_button">Fees</button>
        <p class="text text-6">|</p>
        <button class="_button">Terms &amp; Conditions</button>
        <p class="text text-7">|</p>
        <button class="_button">F.A.Q.</button>
      </div>
    </div>
    <div class="body-container">
      <div class="center-container clearfix">
        <p class="text text-8">Please enter your Nerva wallet for receiving XNV</p>
		<form action="order-status.php" method="post" id="xnv-new-order">
		<input hidden type="text" name="form-type" value="start-order">
        <input class="_input" type="text" name="xnvwallet" required>
        <p class="text text-9">Please enter your Bitcoin wallet for any possible refund</p>
        <input class="_input _input-2" type="text" name="btcrefund" required>
        <div class="re_captcha_container"><center><div class="g-recaptcha" data-sitekey="6LfNi3MUAAAAAB5FnElGDVxn6O8yuKZeFFxOBuVJ"></div></center></div>
		
        <p class="text text-10"><?php if ($_SESSION["captcha"] == "false") { echo "Please verify you are human!"; } 
									else { echo "Please double check the wallet addresses you have entered. Any mistake on the addresses might cause you to lose your funds."; }
									  if ($_SESSION["databasestatus"] == "error") { echo "Database connection failed. Please try again later."; $_SESSION["databasestatus"] = "refreshed"; }
		?></p>
		
        <button class="_button" type="submit" form="xnv-new-order">Start Order</button>
      </div>
    </div>
  </div>
</body>
</html>

order/hiddensrc_exchange/btcrpc.php
<?php



/**
 * Class to facilitate the communication to and from a Bitcoin JSON-RPC server.
 *
 * @author  Liam Williams ( https://github.com/liamwli )
 */
class BitCoin
{
	/**
	 * Connect using secure http.
	 */
	const PROTOCOL_HTTPS = "https";

	/**
	 * Connect using plain http.
	 */
	const PROTOCOL_HTTP = "http";

	/**
	 * Use with @see {setOption}
	 *
	 * Controls the response format. Use constants RESPONSE_ASSOC_ARRAY or RESPONSE_CLASS_FIELDS.
	 */
	const OPTION_RESPONSE_FORMAT = 'optionResponseFormat';

	const RESPONSE_ASSOC_ARRAY  = 'assocArray';
	const RESPONSE_CLASS_FIELDS = 'classFields';

	protected $_options = array();

	protected $_user;
	protected $_password;
	protected $_host;
	protected $_port;

	protected $_protocol = self::PROTOCOL_HTTP;
	protected $_cert     = null;

	protected $_status;
	protected $_error;
	protected $_errorCode;
	protected $_response;
	protected $_rawResponse;

	protected $_id = 0;

	/**
	 * @param string $user     Bitcoin-RPC username
	 * @param string $password Bitcoin-RPC password
	 * @param string $host     Bitcoin host (default 127.0.0.1)
	 * @param string $port     Bitcoin port (8332)
	 *
	 * @throws \Exception
	 */
	public function __construct($user, $password, $host = 'HOST IP', $port = 'PORT')
	{
		if (!extension_loaded('curl'))
		{
			throw new \Exception("Curl is not active on this PHP installation. Please install the curl PHP library.");
		}

		$this->_user = $user;
		$this->_password = $password;
		$this->_host = $host;
		$this->_port = $port;
	}

	/**
	 * Set the protocol used to connect to the JSON-RPC server.
	 *
	 * @param string $protocol The protocol to use. The PROTOCOL_ constants should be used.
	 *
	 * @return $this This object for method chaining.
	 * @throws BitCoinException Thrown if invalid protocol value passed.
	 * @deprecated JSON-RPC over SSL has been deprecated and will be removed from newer bitcoin-core versions.
	 */
	public function setProtocol($protocol)
	{
		switch ($protocol)
		{
			case self::PROTOCOL_HTTPS:
				$this->_protocol = "https";
				break;
			case self::PROTOCOL_HTTP:
				$this->_protocol = "http";
				break;
			default:
				throw new BitCoinException("Invalid Protocol");
		}

		return $this;
	}

	/**
	 * Set the certificate to be used to confirm a HTTPS connection.
	 *
	 * This will set the protocol to {@see PROTOCOL_SSL}.
	 *
	 * @param $filePath
	 *
	 * @return $this This object for method chaining.
	 * @throws BitCoinException
	 * @deprecated JSON-RPC over SSL has been deprecated and will be removed from newer bitcoin-core versions.
	 */
	public function setCertificate($filePath)
	{
		if (!file_exists(realpath($filePath)))
		{
			throw new BitCoinException("Invalid Certificate Path");
		}

		$this->_cert = realpath($filePath);
		$this->setProtocol(self::PROTOCOL_HTTPS);

		return $this;
	}

	/**
	 * Set an option.
	 *
	 * @param $option
	 * @param $optionValue
	 *
	 * @return $this This object for method chaining.
	 */
	public function setOption($option, $optionValue)
	{
		$this->_options[$option] = $optionValue;

		return $this;
	}

	/**
	 * Gets an option value. If option isn't set, option value is set to $defaultValue and returned.
	 *
	 * @param $option
	 * @param $defaultValue
	 *
	 * @return mixed
	 */
	public function getOption($option, $defaultValue)
	{
		if (!isset($this->_options[$option]))
		{
			$this->setOption($option, $defaultValue);
		}

		return $this->_options[$option];
	}

	/**
	 * Call a method and return the response as an associative array.
	 *
	 * @param string $methodName The method to call.
	 * @param array  $parameters Parameters for the method.
	 *
	 * @return array|false
	 * @throws BitCoinException
	 */
	public function callMethod($methodName, array $parameters = array())
	{
		$this->_status = null;
		$this->_response = null;
		$this->_rawResponse = null;
		$this->_error = null;

		$parameters = array_values($parameters);

		$this->_id++;

		$input = json_encode(array(
			'method' => $methodName,
			'params' => $parameters,
			'id' => $this->_id
		));

		$curlResource = curl_init("{$this->_protocol}://{$this->_user}:{$this->_password}@{$this->_host}:{$this->_port}");
		$curlOptions = array(
			CURLOPT_RETURNTRANSFER => true,
			CURLOPT_FOLLOWLOCATION => true,
			CURLOPT_HTTPHEADER => array("Content-type: application/json"),
			CURLOPT_POST => true,
			CURLOPT_POSTFIELDS => $input,
		);

		if ($this->_protocol == 'https')
		{
			if ($this->_cert != null)
			{
				$curlOptions[CURLOPT_CAINFO] = $this->_cert;
				$curlOptions[CURLOPT_CAPATH] = dirname($this->_cert);
				$curlOptions[CURLOPT_SSL_VERIFYPEER] = true;
				$curlOptions[CURLOPT_SSL_VERIFYHOST] = true;
			}
			else
			{
				$curlOptions[CURLOPT_SSL_VERIFYPEER] = false;
				$curlOptions[CURLOPT_SSL_VERIFYHOST] = false;
			}
		}

		curl_setopt_array($curlResource, $curlOptions);

		$this->_rawResponse = curl_exec($curlResource);

		$this->_response = json_decode($this->_rawResponse,
			$this->getOption(self::OPTION_RESPONSE_FORMAT, self::RESPONSE_ASSOC_ARRAY) == self::RESPONSE_ASSOC_ARRAY);

		$this->_status = curl_getinfo($curlResource, CURLINFO_HTTP_CODE);

		$curlError = curl_error($curlResource);

		curl_close($curlResource);

		if ($curlError)
		{
			$this->_error = $curlError;

			return false;
		}

		if ($this->_response['error'])
		{
			$this->_error = $this->_response['error']['message'];
			$this->_errorCode = $this->_response['error']['code'];
		}
		else if ($this->_status != 200)
		{
			switch ($this->_status)
			{
				case 400:
					$this->_error = "HTTP_BAD_REQUEST";
					break;
				case 401:
					$this->_error = "HTTP_UNAUTHORIZED";
					break;
				case 403:
					$this->_error = "HTTP_FORBIDDEN";
					break;
				case 404:
					$this->_error = "HTTP_NOT_FOUND";
					break;
				default:
					throw new BitCoinException("An unexpected http response code was encountered!");
			}
		}

		if ($this->_error)
		{
			return false;
		}

		return $this->_response['result'];
	}

	/**
	 * @return string
	 */
	public function getError()
	{
		return $this->_error;
	}

	/**
	 * @return string
	 */
	public function getRawResponse()
	{
		return $this->_rawResponse;
	}

	/**
	 * @return array
	 */
	public function getResponse()
	{
		return $this->_response;
	}

	/**
	 * @return string
	 */
	public function getStatus()
	{
		return $this->_status;
	}

	/**
	 * @see callMethod
	 */
	function __call($name, $arguments)
	{
		return $this->callMethod(strtolower($name), $arguments);
	}
}

/**
 * General exception for bitcoin-related issues. (Requires message).
 */
class BitCoinException extends \Exception
{
	public function __construct($message, $code = 0, \Exception $previous = null)
	{
		parent::__construct($message, $code, $previous);
	}
}

order/hiddensrc_exchange/xnvrpc.php
<?php



/**
 * Class to facilitate the communication to and from a Bitcoin JSON-RPC server.
 *
 * @author  Liam Williams ( https://github.com/liamwli )
 */
class NervaXNV
{
	/**
	 * Connect using secure http.
	 */
	const PROTOCOL_HTTPS = "https";

	/**
	 * Connect using plain http.
	 */
	const PROTOCOL_HTTP = "http";

	/**
	 * Use with @see {setOption}
	 *
	 * Controls the response format. Use constants RESPONSE_ASSOC_ARRAY or RESPONSE_CLASS_FIELDS.
	 */
	const OPTION_RESPONSE_FORMAT = 'optionResponseFormat';

	const RESPONSE_ASSOC_ARRAY  = 'assocArray';
	const RESPONSE_CLASS_FIELDS = 'classFields';

	protected $_options = array();

	protected $_user;
	protected $_password;
	protected $_host;
	protected $_port;

	protected $_protocol = self::PROTOCOL_HTTP;
	protected $_cert     = null;

	protected $_status;
	protected $_error;
	protected $_errorCode;
	protected $_response;
	protected $_rawResponse;

	protected $_id = 0;

	/**
	 * @param string $user     Bitcoin-RPC username
	 * @param string $password Bitcoin-RPC password
	 * @param string $host     Bitcoin host (default 127.0.0.1)
	 * @param string $port     Bitcoin port (8332)
	 *
	 * @throws \Exception
	 */
	public function __construct($user, $password, $host = '104.248.138.175', $port = '31541')
	{
		if (!extension_loaded('curl'))
		{
			throw new \Exception("Curl is not active on this PHP installation. Please install the curl PHP library.");
		}

		$this->_user = $user;
		$this->_password = $password;
		$this->_host = $host;
		$this->_port = $port;
	}

	/**
	 * Set the protocol used to connect to the JSON-RPC server.
	 *
	 * @param string $protocol The protocol to use. The PROTOCOL_ constants should be used.
	 *
	 * @return $this This object for method chaining.
	 * @throws BitCoinException Thrown if invalid protocol value passed.
	 * @deprecated JSON-RPC over SSL has been deprecated and will be removed from newer bitcoin-core versions.
	 */
	public function setProtocol($protocol)
	{
		switch ($protocol)
		{
			case self::PROTOCOL_HTTPS:
				$this->_protocol = "https";
				break;
			case self::PROTOCOL_HTTP:
				$this->_protocol = "http";
				break;
			default:
				throw new BitCoinException("Invalid Protocol");
		}

		return $this;
	}

	/**
	 * Set the certificate to be used to confirm a HTTPS connection.
	 *
	 * This will set the protocol to {@see PROTOCOL_SSL}.
	 *
	 * @param $filePath
	 *
	 * @return $this This object for method chaining.
	 * @throws BitCoinException
	 * @deprecated JSON-RPC over SSL has been deprecated and will be removed from newer bitcoin-core versions.
	 */
	public function setCertificate($filePath)
	{
		if (!file_exists(realpath($filePath)))
		{
			throw new BitCoinException("Invalid Certificate Path");
		}

		$this->_cert = realpath($filePath);
		$this->setProtocol(self::PROTOCOL_HTTPS);

		return $this;
	}

	/**
	 * Set an option.
	 *
	 * @param $option
	 * @param $optionValue
	 *
	 * @return $this This object for method chaining.
	 */
	public function setOption($option, $optionValue)
	{
		$this->_options[$option] = $optionValue;

		return $this;
	}

	/**
	 * Gets an option value. If option isn't set, option value is set to $defaultValue and returned.
	 *
	 * @param $option
	 * @param $defaultValue
	 *
	 * @return mixed
	 */
	public function getOption($option, $defaultValue)
	{
		if (!isset($this->_options[$option]))
		{
			$this->setOption($option, $defaultValue);
		}

		return $this->_options[$option];
	}

	/**
	 * Call a method and return the response as an associative array.
	 *
	 * @param string $methodName The method to call.
	 * @param array  $parameters Parameters for the method.
	 *
	 * @return array|false
	 * @throws BitCoinException
	 */
	public function callMethod($methodName, array $parameters = array())
	{
		$this->_status = null;
		$this->_response = null;
		$this->_rawResponse = null;
		$this->_error = null;

		$parameters = array_values($parameters);

		$this->_id++;

		$input = json_encode(array(
			'method' => $methodName,
			'params' => $parameters,
			'id' => $this->_id
		));

		$curlResource = curl_init("{$this->_protocol}://{$this->_user}:{$this->_password}@{$this->_host}:{$this->_port}");
		$curlOptions = array(
			CURLOPT_RETURNTRANSFER => true,
			CURLOPT_FOLLOWLOCATION => true,
			CURLOPT_HTTPHEADER => array("Content-type: application/json"),
			CURLOPT_POST => true,
			CURLOPT_POSTFIELDS => $input,
		);

		if ($this->_protocol == 'https')
		{
			if ($this->_cert != null)
			{
				$curlOptions[CURLOPT_CAINFO] = $this->_cert;
				$curlOptions[CURLOPT_CAPATH] = dirname($this->_cert);
				$curlOptions[CURLOPT_SSL_VERIFYPEER] = true;
				$curlOptions[CURLOPT_SSL_VERIFYHOST] = true;
			}
			else
			{
				$curlOptions[CURLOPT_SSL_VERIFYPEER] = false;
				$curlOptions[CURLOPT_SSL_VERIFYHOST] = false;
			}
		}

		curl_setopt_array($curlResource, $curlOptions);

		$this->_rawResponse = curl_exec($curlResource);

		$this->_response = json_decode($this->_rawResponse,
			$this->getOption(self::OPTION_RESPONSE_FORMAT, self::RESPONSE_ASSOC_ARRAY) == self::RESPONSE_ASSOC_ARRAY);

		$this->_status = curl_getinfo($curlResource, CURLINFO_HTTP_CODE);

		$curlError = curl_error($curlResource);

		curl_close($curlResource);

		if ($curlError)
		{
			$this->_error = $curlError;

			return false;
		}

		if ($this->_response['error'])
		{
			$this->_error = $this->_response['error']['message'];
			$this->_errorCode = $this->_response['error']['code'];
		}
		else if ($this->_status != 200)
		{
			switch ($this->_status)
			{
				case 400:
					$this->_error = "HTTP_BAD_REQUEST";
					break;
				case 401:
					$this->_error = "HTTP_UNAUTHORIZED";
					break;
				case 403:
					$this->_error = "HTTP_FORBIDDEN";
					break;
				case 404:
					$this->_error = "HTTP_NOT_FOUND";
					break;
				default:
					throw new BitCoinException("An unexpected http response code was encountered!");
			}
		}

		if ($this->_error)
		{
			return false;
		}

		return $this->_response['result'];
	}

	/**
	 * @return string
	 */
	public function getError()
	{
		return $this->_error;
	}

	/**
	 * @return string
	 */
	public function getRawResponse()
	{
		return $this->_rawResponse;
	}

	/**
	 * @return array
	 */
	public function getResponse()
	{
		return $this->_response;
	}

	/**
	 * @return string
	 */
	public function getStatus()
	{
		return $this->_status;
	}

	/**
	 * @see callMethod
	 */
	function __call($name, $arguments)
	{
		return $this->callMethod(strtolower($name), $arguments);
	}
}

/**
 * General exception for bitcoin-related issues. (Requires message).
 */
class BitCoinException extends \Exception
{
	public function __construct($message, $code = 0, \Exception $previous = null)
	{
		parent::__construct($message, $code, $previous);
	}
}

order-status.php
<?php session_start(); 

$post_data = http_build_query(
    array(
        'secret' => "RECAPTCHA SECRET",
        'response' => $_POST['g-recaptcha-response'],
        'remoteip' => $_SERVER['REMOTE_ADDR']
    )
);
$opts = array('http' =>
    array(
        'method'  => 'POST',
        'header'  => 'Content-type: application/x-www-form-urlencoded',
        'content' => $post_data
    )
);
$context  = stream_context_create($opts);
$response = file_get_contents('https://www.google.com/recaptcha/api/siteverify', false, $context);
$result = json_decode($response);

if (!$result->success) {
	$_SESSION["captcha"] = "false";
	
	if ($_POST["form-type"] == "check-order") { header( "Location: index.php" ); } 
	elseif ($_POST["form-type"] == "start-order") { header( "Location: start-order.php" ); }
	else { header ( "Location: index.php" ); }
}

else { $_SESSION["captcha"] = "valid"; }



include('order/hiddensrc_exchange/btcrpc.php');

$servername = "localhost";
$username = "USERNAME";
$password = "PASSWORD";
$db_name = "DB NAME";


$conn = new mysqli($servername, $username, $password, $db_name);

if ($conn->connect_error) {
$_SESSION["databasestatus"] = "error";
header( "Location: start-order.php" );
} 

$bitcoin = new BitCoin('BTC RPC USERNAME','PASSWORD');

if ($_POST["form-type"] == "check-order") { 
$orderid = $_POST["order_id_check"];
$sql = "SELECT * FROM exchange WHERE order_id = \"". $orderid . "\"";
$results = mysqli_query($conn,$sql);
if (mysqli_num_rows($results) == 0) {
$order_status = 3; 
$_SESSION["error_status"] = "no_order_with_given_id";
header ( "Location: index.php" );
}
else { $order_status = 2; 
$inforow = mysqli_fetch_array($results);
$btcwallet = $inforow[3];
$btcrefund = $inforow[4];
$nervadeposit = $inforow[5];
$btc_status = $inforow[6];
$nerva_status = $inforow[7];
$nerva_sent_amount = $inforow[8];
$btc_deposit_amount = $inforow[9];
$nervatxhash = $inforow[10];}

 }
else if ($_SESSION["captcha"] == "valid" ) {
$orderid = substr(md5(microtime()),rand(0,26),12); 
$btcrefund = $_POST["btcrefund"];
$nervadeposit = $_POST["xnvwallet"];
$btcwallet = $bitcoin->getaccountaddress($orderid);
$sql = "INSERT INTO exchange (order_id, btc_deposit, btc_refund, nerva_deposit, btc_deposit_status, nerva_sent_status, nerva_sent_amount, btc_deposit_amount, nerva_tx_hash)
VALUES ('$orderid','$btcwallet','$btcrefund','$nervadeposit','-1','0','0','0','0')";
$btc_status = -1;
if ($conn->query($sql) === TRUE) {
    $order_status = 1;
} else {
    $order_status = 0;
	$error_status = "order_creation_failed_database_error";
}}





$qrurl = "https://api.qrserver.com/v1/create-qr-code/?data=" . $btcwallet . "&size=150x150";
?>
<html>
<head>
<link rel="shortcut icon" href="/favicon.ico" />
  <meta charset="utf-8">
  <meta name="viewport" content="initial-scale=1.0">
  <title>Nerva.Exchange | Order Status</title>
  <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,300|Montserrat:400,400|Raleway:400,100,700" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="css/standardize.css">
  <link rel="stylesheet" href="css/order-status-grid.css">
  <link rel="stylesheet" href="css/order-status.css">
</head>
<body class="body page-order-status clearfix">
  <div class="wholebody clearfix">
    <div class="header clearfix">
      <div class="logo">
        <p class="website-name"><strong>Nerva.Exchange</strong></p>
      </div>
      <div class="header-links clearfix">
        <button id="home_button" class="_button _button-1">Home</button>
        <p class="text">|</p>
        <button id="aboutus_button" class="_button _button-2">About Us</button>
        <p class="text">|</p>
        <button id="fees_button" class="_button _button-3">Fees</button>
      </div>
    </div>
    <div class="body-container clearfix">
      <div class="container container-1 clearfix">
        <div class="container container-2 clearfix">
          <p class="text text-3">Order ID : <?php echo $orderid; ?></p>
          <p class="text text-4">Order ID is important for us to track your order in case of any problem. If you lose your Order ID and have a problem, you might lose your funds.</p>
          <p class="text text-5">Nerva (XNV) Deposit Address</p>
          <p class="text text-6"><?php echo $nervadeposit; ?></p>
          <p class="text text-7">Bitcoin (BTC) Refund Address</p>
          <p class="text text-8"><?php echo $btcrefund;;?></p>
        </div>
        <div class="container container-3 clearfix">
          <p class="text text-9">You have a problem with the exchange?</p>
          <button class="_button _button-4">REACH US</button>
        </div>
      </div>
      <div class="container container-4 clearfix">
        <p class="text text-10">Order Status</p>
		<?php if ($btc_status == -1) { 
        echo "<p class=\"text text-11\">Waiting for Bitcoin Deposit</p>";
        echo "<div class=\"new-order clearfix\">";
          echo "<div class=\"qr-code\"><img style=\"-webkit-user-select: none;\" src=\"" . $qrurl . "\"></div>";
          echo "<p class=\"text text-12\">" . $btcwallet . "</p>";
          echo "<p class=\"text text-13\">Please deposit Bitcoin to the address above</p>";
          echo "<p class=\"text text-14\">Minimum Deposit Amount is : 0.00015</p>";
          echo "<p class=\"text text-15\">Maximum Deposit Amount is : 0.00200</p>";
        echo "</div>"; }
		
			else if ($btc_status == 0 || $btc_status == 1 || $btc_status == 2 ) { 
			echo "<p class=\"text text-11\">Waiting for confirmations from network</p>"; }
			
			else if ($btc_status == 3) {
			echo "<p class=\"text text-11\">Your exchange has been completed!</p>";
			echo "<div class=\"new-order clearfix\">";
				echo "<p class=\"text text-12\">" . "Your final exchange rate was : " . number_format($btc_deposit_amount / $nerva_sent_amount, 10) . "</p>";
				echo "<p class=\"text text-13\">Total ". $nerva_sent_amount . " XNV sent</p>";
				echo "<p class=\"text text-14\">The exchange rate includes all conversion and transfer fees.</p>";
				echo "<p class=\"text text-15\">TX Hash : XXXXXXX</p>";
			echo "</div>"; }
	  ?>
	  </div>
    </div>
	<div class="container container-5 clearfix">
        <div class="container container-6 clearfix">
          <p class="text text-16">Nerva.Exchange</p>
          <p class="text text-17">© 2018 All Rights Reserved</p>
        </div>
        <div class="container container-7 clearfix">
          <button class="_button _button-5">About Us</button>
          <p class="text text-18">|</p>
          <button class="_button _button-6">Fees</button>
          <p class="text text-19">|</p>
          <button class="_button _button-7">Terms &amp; Conditions</button>
          <p class="text text-20">|</p>
          <button class="_button _button-8">F.A.Q.</button>
        </div>
      </div>
  </div>
</body>
</html>